dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentRelationshipsHelper.java

```diff
@@ -0,0 +1,186 @@
+    }
+
+    /**
+     * Returns a singleton instance of this class.
+     *
+     * @return A single instance of this class.
+     */
+    public static ContentRelationshipsHelper getInstance() {
+        return ContentRelationshipsHelper.SingletonHolder.INSTANCE;
+    }
+
+    /**
+     * Returns a list of contents and its relationships according to a given set of parameters.
+     *
+     * @param request   The {@link HttpServletRequest} object.
+     * @param user      The {@link User} performing this action.
+     * @param paramsMap A Map of parameters that will define the search criteria.
+     * @return The list of associated contents.
+     */
+    public List<Contentlet> getRelatedContent(final HttpServletRequest request, final User user,
+                                              final Map<String, String> paramsMap) {
+        final String query = paramsMap.get(RESTParams.QUERY.getValue());
+        final String id = paramsMap.get(RESTParams.ID.getValue());
+        String orderBy = paramsMap.get(RESTParams.ORDERBY.getValue());
+        final String limitStr = paramsMap.get(RESTParams.LIMIT.getValue());
+        final String offsetStr = paramsMap.get(RESTParams.OFFSET.getValue());
+        final String inode = paramsMap.get(RESTParams.INODE.getValue());
+        orderBy = UtilMethods.isSet(orderBy) ? orderBy : "modDate desc";
+        long language = this.languageAPI.getDefaultLanguage().getId();
+
+        if (null != paramsMap.get(RESTParams.LANGUAGE.getValue())) {
+            try {
+                language = Long.parseLong(paramsMap.get(RESTParams.LANGUAGE.getValue()));
+            } catch (Exception e) {
+                Logger.warn(this.getClass(), "The specified language ID is invalid. Using the " +
+                        "default language ID.");
+            }
+        }
+        // Limit and Offset Parameters Handling, if not passed, using default
+        int limit = 10;
+        int offset = 0;
+        try {
+            if (UtilMethods.isSet(limitStr)) {
+                limit = Integer.parseInt(limitStr);
+            }
+        } catch (NumberFormatException e) {
+            // Could not parse limit value. Use default value
+        }
+        try {
+            if (UtilMethods.isSet(offsetStr)) {
+                offset = Integer.parseInt(offsetStr);
+            }
+        } catch (NumberFormatException e) {
+            // Could not parse offset value. Use default value
+        }
+        final boolean live = (null == paramsMap.get(RESTParams.LIVE.getValue()) || !"false"
+                .equals(paramsMap.get(RESTParams.LIVE.getValue())));
+        // Fetching the content using a query if passed or an id
+        List<Contentlet> contentlets = new ArrayList<>();
+        boolean idPassed = false;
+        boolean inodePassed = false;
+        boolean queryPassed = false;
+        try {
+            if (idPassed = UtilMethods.isSet(id)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment out of this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)